Assets/MixedRealityToolkit/Inspectors/ServiceInspectors/BaseMixedRealityServiceInspector.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using System;
+using UnityEditor;
+
+namespace Microsoft.MixedReality.Toolkit.Editor
+{
+    public class BaseMixedRealityServiceInspector : IMixedRealityServiceInspector
+    {
+        public virtual bool DrawProfileField { get { return true; } }
+
+        public virtual bool AlwaysDrawSceneGUI { get { return false; } }
+
+        public virtual void DrawGizmos(object target) { }
+
+        public virtual void DrawInspectorGUI(object target) { }
+
+        public virtual void DrawSceneGUI(object target, SceneView sceneView) { }
```

Comment:
>public virtual void DrawSceneGUI(object target, SceneView sceneView) { } [](start = 8, length = 72)

Would it also be possible to have some comments regarding the type that "target" will be - I see that it gets cast to some other interesting things in subclasses, so maybe setting some general expectations (i.e. it will be of type [CustomEditor(typeof(ServiceFacade))] whatever is in CustomEditor) or MixedRealityServiceInspector

In general these types of comments are valuable especially with generic object types, to help scope down the set of things consumers/extenders would have to deal with with mental overhead.

Reply:
Actually this is probably just a comment for IMixedRealityServiceInspector's parameters. 

---
In reply to: [278734920](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3773#discussion_r278734920) [](ancestors = 278734920)